File: suitecrm-migration/backend/maps-service/src/main/java/com/suitecrm/maps/service/MapsService.java
package com.suitecrm.maps.service;

import com.suitecrm.maps.dto.*;
import com.suitecrm.maps.entity.*;
import com.suitecrm.maps.repository.*;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.util.UUID;

@Service @RequiredArgsConstructor
public class MapsService {
    private final GeoMapRepository geoMapRepository;
    private final MarkerRepository markerRepository;
    private final AreaRepository areaRepository;

    public Page<GeoMapDto> getAllMaps(Pageable pageable) {
        return geoMapRepository.findByDeletedFalse(pageable).map(this::toMapDto);
    }
    public Page<MarkerDto> getAllMarkers(Pageable pageable) {
        return markerRepository.findByDeletedFalse(pageable).map(this::toMarkerDto);
    }
    public MarkerDto getMarkerById(UUID id) {
        return toMarkerDto(markerRepository.findByIdAndDeletedFalse(id)
                .orElseThrow(() -> new RuntimeException("Marker not found: " + id)));
    }

    private GeoMapDto toMapDto(GeoMap m) {
        return GeoMapDto.builder().id(m.getId()).name(m.getName()).centerLat(m.getCenterLat()).centerLng(m.getCenterLng())
                .zoomLevel(m.getZoomLevel()).mapType(m.getMapType()).moduleType(m.getModuleType())
                .description(m.getDescription()).assignedUserId(m.getAssignedUserId())
                .dateEntered(m.getDateEntered()).dateModified(m.getDateModified()).build();
    }
    private MarkerDto toMarkerDto(Marker m) {
        return MarkerDto.builder().id(m.getId()).name(m.getName()).city(m.getCity()).state(m.getState())
                .country(m.getCountry()).latitude(m.getLatitude()).longitude(m.getLongitude())
                .markerType(m.getMarkerType()).relatedModule(m.getRelatedModule()).relatedId(m.getRelatedId())
                .description(m.getDescription()).assignedUserId(m.getAssignedUserId())
                .dateEntered(m.getDateEntered()).dateModified(m.getDateModified()).build();
    }
}

File: suitecrm-migration/frontend/shell/src/app/services/maps.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface GeoMap {
  id: string;
  name: string;
  centerLat: number;
  centerLng: number;
  zoomLevel: number;
  mapType: string;
  moduleType: string;
}

export interface Marker {
  id: string;
  name: string;
  city: string;
  state: string;
  country: string;
  latitude: number;
  longitude: number;
  markerType: string;
  relatedModule: string;
  relatedId: string;
}

@Injectable({ providedIn: 'root' })
export class MapsService {
  private apiUrl = '/api/maps';

  constructor(private http: HttpClient) {}

  getMaps(page = 0, size = 20): Observable<any> { return this.http.get(`${this.apiUrl}?page=${page}&size=${size}`); }
  getMarkers(page = 0, size = 20): Observable<any> { return this.http.get(`${this.apiUrl}/markers?page=${page}&size=${size}`); }
  getMarkerById(id: string): Observable<Marker> { return this.http.get<Marker>(`${this.apiUrl}/markers/${id}`); }
}
